### PR TITLE
Balder/separate empalce back and emplace

### PR DESCRIPTION
**Pull request by baldersheim**
@havardpe PR

### CODE REVIEW

**Comment by baldersheim** [comment]
@geirst PR

File: vespalib/src/vespa/vespalib/stllike/vector_map.h [hunk 1]
```diff
@@ -82,6 +79,19 @@ void swap(vector_map<K, V, LT> & a, vector_map<K, V, LT> & b)
     a.swap(b);
 }
 
+template< typename K, typename V, typename LT >
+V &
+vector_map<K, V, LT>::operator[](const K &key) {
+    LT lt;
+    iterator f = std::lower_bound(begin(), end(), key, KeyOrder());
+    if (f == end()) {
+        _ht.template emplace_back(key, V());
```
Comment (havardpe) [inline vespalib/src/vespa/vespalib/stllike/vector_map.h:88]: how much slower is emplace at end() compared to emplace_back?

File: vespalib/src/vespa/vespalib/stllike/vector_map.h [hunk 2]
```diff
@@ -28,45 +28,42 @@ class vector_map
...
-    V & operator [] (const K & key)          {
-        value_type v(key, V());
-        LT lt;
-        iterator f = std::lower_bound(begin(), end(), v);
-        if ((f == end()) || lt(key, f->first)) {
-            f = _ht.insert(f, v);
-        }
-        return f->second; 
-    }
+    V & operator [] (const K & key) const    { return find(key)->second; }
```
[the remark this answers is not in the document]
Comment (baldersheim) [reply vespalib/src/vespa/vespalib/stllike/vector_map.h]: Removed, unused.

File: vespalib/src/vespa/vespalib/stllike/vector_map.h (hunk 1 above)
Comment (baldersheim) [reply vespalib/src/vespa/vespalib/stllike/vector_map.h:87]: emplace_back, and push_back has inlined fast path.
emplace, and insert does not.

**Pull request merged**